### PR TITLE
[Lora] Lora quat info re-factor and support deepseekv3 mla lora

### PR DESCRIPTION
**Pull request by yushengsu-thu**
## Modifications

- MoE quant-info refactor
Extracts get_triton_quant_info() into each quantization method (FP8, INT8, WNA16, etc.) so FusedMoEWithLoRA correctly receives quantization scales/flags, enabling LoRA on quantized MoE models.

- DeepSeek-V3 MLA LoRA support
Adds ReplicatedLinearWithLoRA to handle the fused q_a_proj + kv_a_proj_with_mqa projection unique to MLA. Since the two sub-projections have unequal output dimensions, LoRA B is split and applied via two separate sgemm calls. Includes weight fusion, hidden-dim fixes (v_head_dim for o_proj, shared-expert sizes for MoE layers), and TP exclusion for replicated layers.

- CI: Adds a 5-GPU GB200 logprob accuracy test against DeepSeek-V3.1-Base.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/base_config.py [hunk 1]
```diff
@@ -106,6 +107,26 @@ def apply(
...
+        stored on *layer*.
+
+        The LoRA MoE runner calls this so that ``invoke_fused_moe_kernel``
+        receives the correct flags / scales / block-shape for the base
+        weights.  Each quantisation method should override this with the
+        same construction it already uses inside ``apply()``.
+
+        The default returns an unquantised descriptor.
+        """
+        from sglang.srt.layers.moe.moe_runner.triton import TritonMoeQuantInfo
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/base_config.py]: `raise NotImplementedError` here since it's base class

File: test/registered/lora/test_lora_deepseek_v3_base_logprob_diff.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import torch
+from huggingface_hub import snapshot_download
+
+import sglang as sgl
+from sglang.test.ci.ci_register import register_cuda_ci
+from sglang.test.test_utils import CustomTestCase
+
+register_cuda_ci(
+    est_time=300,
+    suite="stage-c-test-4-gpu-b200",
```
Comment (Fridge003) [inline test/registered/lora/test_lora_deepseek_v3_base_logprob_diff.py]: Register to `nightly-8-gpu-b200`
Also please post the result of running this test on 8-gpu b200

File: python/sglang/srt/layers/quantization/base_config.py (hunk 1 above)
Comment (yushengsu-thu) [reply python/sglang/srt/layers/quantization/base_config.py]: done

**Review by Fridge003 — APPROVED** [review]
